### PR TITLE
Change TIMESTAMP partitioning to be completely tz-independent

### LINKED ISSUE #300: Constraint violation for timestamp without timezone
OS: FreeBSD 11.1
PostgreSQL: 9.6.6
TimescaleDB: 0.6.0

I was copying data from a regular table into a TimescaleDB table via `INSERT... SELECT... WHERE ts >= $1 AND ts < $2` and got the following error:

```
ERROR:  new row for relation "_hyper_1_40_chunk" violates check constraint "constraint_15"
DETAIL:  Failing row contains (31485671542, 2017-09-21 19:00:00, 545, 777710, 'z').
```

I tracked down the constraint to this expression:
```sql
(("ts" >= '2017-09-22 00:00:00'::timestamp without time zone)
 AND ("ts" < '2017-09-23 00:00:00'::timestamp without time zone))
```

I'll also note that my server's time zone is CST (currently UTC-05:00). Both tables define `ts` as `timestamp without time zone`, and when I select from the source table, I indeed see the timestamp exactly as it appears in the error message. Where it gets weird is, I can single out this particular row and insert it just fine:

```sql
INSERT INTO process SELECT * FROM old_process WHERE id = 31485671542;
```

What exactly is going on here? Is TimescaleDB using the server time zone to determine which chunk the row goes into? I don't understand why my second statement works just fine when the only difference is the number of rows returned by the `SELECT`.

Also: a feature request. In the future, can the constraint creation process include the column name in the constraint name? Possibly the chunk name as well, since the constraint names end up repeating a lot. It would be much easier to track down the problem if I don't have to filter out a dozen constraints with identical names.

### PR DESCRIPTION
**Pull request by cevian**
Previously, for timezones w/o tz. The range_end and range_start were
defined as UTC, but the constraints on the table were written as as
the local time at the time of chunk creation. This does not work well
if timezones change over the life of the hypertable.

This change removes the dependency on local time for all timestamp
partitioning. Namely, the range_start and range_end remain as UTC
but the constraints are now always written in UTC too. Since old
constraints correctly describe the data currently in the chunks, the
update script to handle this change changes range_start and range_end
instead of the constraints.

Fixes #300.

### CODE REVIEW

File: test/expected/ddl_alter_column.out [hunk 1]
```diff
@@ -61,7 +61,10 @@ SELECT * FROM alter_test WHERE time > '2017-05-20T10:00:01';
 
 -- rename column and change its type
 ALTER TABLE alter_test RENAME COLUMN time TO time_us;
+--converting timestamptz->timestamp should happen under UTC
+SET timezone = 'UTC';
```
Comment (erimatnor) [inline test/expected/ddl_alter_column.out]: What does this mean? That it is only valid to convert timestamptz if TZ is UTC? Should we even allow such conversions in that case? Or, is this strictly something required only for our tests?

Comment (cevian) [reply test/expected/ddl_alter_column.out]: Yeah for now I think it's ok to leave this as valid only under UTC. Didn't want to put in too much work here as this is a weird edge case but wanted to keep consistent for tests. I think it's ok not to block since you get a constraint violation error anyway if you do something wrong.

Comment (erimatnor) [reply test/expected/ddl_alter_column.out]: Can't we just add a check in `process_utility` that the timezone is UTC and raise an error otherwise?
